apps/api/src/applicant/external-api.service.ts

```diff
@@ -462,36 +495,56 @@ export class ExternalAPIService {
         if (item.hasOwnProperty('milestones') && Array.isArray(item.milestones)) {
           const existingMilestones = item.milestones;
           existingMilestones.forEach(m => {
-            if (allowedMilestones.includes(m.id)) {
-              const temp: any = {
-                status: m.id,
-                start_date: m.start_date,
-                created_date: m.created_date,
-                updated_date: m.created_date,
-                applicant: tableId,
-                notes: m?.note,
-                added_by: null,
-              };
-              if ('added_by' in m && m.added_by > 0 && users[m.added_by]) {
-                temp['added_by'] = m.added_by;
-              }
-              if ('reason_id' in m) {
-                temp['reason'] = m.reason_id;
-              }
-              if ('reason_other' in m) {
-                temp['reason_other'] = m.reason_other;
-              }
-              if ('effective_date' in m) {
-                temp['effective_date'] = m.effective_date;
-              }
-              milestones.push(temp);
-            } else {
-              this.logger.log(`rejected Id ${m.id}`);
-            }
+            this.mapMilestoneData(allowedMilestones, m, milestones, tableId, users);
           });
         }
       },
     );
     return milestones;
   }
+
+  /** create applicant-milestone object */
+  mapMilestoneData(
+    allowedMilestones: string | any[],
+    m: {
+      id: any;
+      start_date: any;
+      created_date: any;
+      note: any;
```

Comment:
how come 'any'?

Reply:
Auto ts lint, picks any if it doesn’t know its source type. In this case it’s coming from ATS and read as a json object. I can fix few any, let me do that.